cmd/basin/publication.go

```diff
@@ -456,6 +464,56 @@ func newPublicationDealsCommand() *cli.Command {
 	}
 }
 
+func newPublicationRetrieveCommand() *cli.Command {
+	return &cli.Command{
+		Name:  "retrieve",
+		Usage: "Retrieve files by CID",
+		Action: func(cCtx *cli.Context) error {
+			arg := cCtx.Args().Get(0)
+			if arg == "" {
+				return errors.New("argument is empty")
+			}
+
+			rootCid, err := cid.Parse(arg)
+			if err != nil {
+				return errors.New("cid is invalid")
+			}
+
+			lassie, err := lassie.NewLassie(cCtx.Context)
+			if err != nil {
+				return fmt.Errorf("failed to create lassie instance: %s", err)
+			}
+
+			carOpts := []car.Option{
+				car.WriteAsCarV1(true),
+				car.StoreIdentityCIDs(false),
+				car.UseWholeCIDs(false),
+			}
+			carWriter := deferred.NewDeferredCarWriterForPath(fmt.Sprintf("./%s.car", arg), []cid.Cid{rootCid}, carOpts...)
```

Comment:
it simply writes the car file into the current dir with CID as its name. to extract gotta use [car extract](https://github.com/ipld/go-car)